include/fiction/algorithms/simulation/sidb/determine_electrostatic_influence_region.hpp

```diff
@@ -0,0 +1,285 @@
+    {
+        const auto simulation_results = quickexact(sublayout, quickexact_parameter);
+        // find the ground state, which is the charge distribution with the lowest energy
+        const auto ground_state_energy = minimum_energy(simulation_results.charge_distributions);
+        for (const auto& lyt : simulation_results.charge_distributions)
+        {
+            if (std::fabs(lyt.get_system_energy() - ground_state_energy) < std::numeric_limits<double>::epsilon())
+            {
+                ground_state_charge_indices_of_sublyt.insert(lyt.get_charge_index_and_base().first);
+            }
+        }
+    }
+    /**
+     * It determines if every cell in the sublayout is part of the layout.
+     *
+     * @tparam Lyt The type of the layout.
+     * @return `true` if the sublayout is a subset of the layout, `false` otherwise.
+     */
+    [[nodiscard]] bool is_sublayout_of_layout() const noexcept
+    {
+        std::vector<typename Lyt::cell> all_sidbs = {};
+        all_sidbs.reserve(layout.num_cells());
+        layout.foreach_cell([&all_sidbs](const auto& c1) { all_sidbs.push_back(c1); });
+
+        std::vector<typename Lyt::cell> all_sidbs_sublayout = {};
+        all_sidbs_sublayout.reserve(layout.num_cells());
+        sublayout.foreach_cell([&all_sidbs_sublayout](const auto& c1) { all_sidbs_sublayout.push_back(c1); });
+
+        for (const auto& cell : all_sidbs_sublayout)
+        {
+            auto it = std::find(all_sidbs.begin(), all_sidbs.end(), cell);
+
+            if (it == all_sidbs.end())
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+    /**
+     * Complete Layout.
+     */
+    Lyt layout{};
+    /**
+     * Sublayout of layout.
+     */
+    Lyt sublayout{};
+    /**
+     * Simulation parameters.
+     */
+    quickexact_params<cell<Lyt>> quickexact_parameter{};
+    /**
+     * Charge indices of the ground state of the sublayout without neighboring SiDBs, i.e., undisturbed.
+     */
+    std::set<uint64_t> ground_state_charge_indices_of_sublyt{};
+    /**
+     * All cells of the layout, excluding the cells of the sublayout.
+     */
+    std::set<typename Lyt::cell> cells_without_sublayout_cells{};
+};
+
+}  // namespace detail
+
+/**
+ * This function identifies the influence zone for a given sublayout within a larger layout. The influence zone
+ * encompasses the neighboring SiDBs whose charge states can potentially affect the ground state of the sublayout.
+ * SiDBs outside the influence zone are considered to have no significant impact on the sublayout's ground state,
+ * and their charge states do not need to be considered during the simulation of the sublayout.
+ *
+ * @tparam Lyt The type representing the layout.
+ * @param layout Layout which includes the sublayout.
+ * @param sublayout Sublayout whose influence region is simulated and which is part of the layout.
+ * @param params *QuickExact* parameters for the physical simulation.
```

Comment:
Even if you don't have any other parameters than QuickExact's (yet), I would still wrap it into a `determine_electrostatic_influence_region_params` helper `struct` in case you want to add some parameters later. It would prevent you from having to change the API, which is always a problem.

Reply:
You are right! Thank you!